File: Problems/0201.-0300./0242._Valid_Anagram/ValidAnagram0.py
#!/bin/python3
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        # Dictionary to hold chars in s
        dict_s = {}

        for char in s:
            if char not in dict.keys(dict_s):
                dict_s[char] = 0
            else:
                dict_s[char] += 1

        # Dictionary to hold chars in t
        dict_t = {}

        for char in t:
            if char not in dict.keys(dict_t):
                dict_t[char] = 0
            else:
                dict_t[char] += 1
        return dict_s == dict_t

File: Problems/0201.-0300./0242._Valid_Anagram/ValidAnagram0.java
//!/bin/java
import java.util.HashMap;

class Solution {
    public boolean isAnagram(String s, String t) {
        // Create HashMap object to hold characters and frequencies
        HashMap<Character, Integer> dict_s = new HashMap<Character, Integer>();
        int count_s = 0;

        for (int i = 0; i < s.length(); i++) {
            if (dict_s.get(s.charAt(i)) == null) {
                dict_s.put(s.charAt(i), 0);
            } else {
                count_s++;
                dict_s.put(s.charAt(i), count_s);
            }
        }

        HashMap<Character, Integer> dict_t = new HashMap<Character, Integer>();
        int count_t = 0;

        for (int j = 0; j < t.length(); j++) {
            if (dict_t.get(t.charAt(j)) == null) {
                dict_t.put(t.charAt(j), 0);
            } else {
                count_t++;
                dict_t.put(t.charAt(j), count_t);
            }
        }

        return dict_t.equals(dict_s);
    }
}
